server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactRange.java

```diff
@@ -83,7 +87,23 @@ public long isReady(FateId fateId, Manager env) throws Exception {
   @Override
   public Repo<Manager> call(final FateId fateId, Manager env) throws Exception {
     CompactionConfigStorage.setConfig(env.getContext(), fateId, config);
-    return new CompactionDriver(namespaceId, tableId, startRow, endRow);
+    BulkImport.KeyExtentCache extentCache = new ConcurrentKeyExtentCache(tableId, env.getContext());
+    KeyExtent keyExtent = extentCache.lookup(new Text(startRow));
```

Comment:
Should not do a lookup if startRow is null.  Also not sure we should use ConcurrentKeyExtentCache here as it will read one hundred extents into memory because its anticipating lots of subsequent lookups.  However in this case there is only one other look and if it misses what the first cached, then it will read another 100 extents into memory.

Instead of using ConcurrentKeyExtentCache, could do the following instead to lookup a single extent that contains the row.

```java
  private static KeyExtent findContaining(Ample ample, TableId tableId, Text row){
    Objects.requireNonNull(row);
    try(var tablets = ample.readTablets().forTable(tableId).overlapping(row, true, row).fetch(PREV_ROW).build()){
      return Iterables.getOnlyElement(tablets).getExtent();
    }
  }
```